app/infrastructure/firebase/services/tenant_init_firestore.py

```
@@ -0,0 +1,202 @@
+            try:
+                await self._client.batch_write(delete_writes)
+                logger.warning(
+                    "Rollback completed for tenant %s (%s documents deleted)",
+                    tenant_id,
+                    len(delete_writes),
+                )
+            except Exception as rollback_exc:
+                logger.exception(
+                    "Rollback failed for tenant %s; tenant may be partially initialized: %s",
+                    tenant_id,
+                    rollback_exc,
```

Ruff (0.15.0)

[warning] Redundant exception object included in `logging.exception` call

(TRY401)